p4c_bm/gen_json.py

```diff
@@ -1199,6 +1211,39 @@ def dump_field_aliases(json_dict, hlir, path_field_aliases):
     json_dict["field_aliases"] = field_aliases
 
 
+def dump_extern_instances(json_dict, hlir):
+    extern_instances = []
+    id_ = 0
+    for name, p4_extern_instance in hlir.p4_extern_instances.items():
+        extern_instance_dict = OrderedDict()
+        extern_instance_dict["name"] = name
+        extern_instance_dict["id"] = id_
+        extern_instance_dict["type"] = p4_extern_instance.extern_type.name
+
+        id_ += 1
+
+        attributes = []
+        for attribute, attr in p4_extern_instance.attributes.items():
+            attr_type = p4_extern_instance.extern_type.attributes[attribute].\
+                        value_type.type_name
+            if attr_type != "bit" and attr_type != "int":
```

Comment:
You add it at the end of the if statement (2 whitespaces, followed with `# pragma: no cover`). This will exclude the if block in coverage measurements. I don't think your program covers this code, which emits a critical error log message. We currently don't have a way to check that invalid programs are handled correctly, and your P4 program is correct because it uses `int` as the attribute type.